include/etcpal/cpp/thread.h

```diff
@@ -227,7 +231,11 @@ inline Thread::Thread(Function&& func, Args&&... args)
 inline Thread::~Thread()
 {
   if (thread_)
+#if ETCPAL_THREAD_JOIN_CAN_TIMEOUT
+    etcpal_thread_timed_join(thread_.get(), max_shutdown_time_);
```

Comment:
This can just always use etcpal_thread_timed_join(), after my other comment about making it available on every platform is implemented.